### PR TITLE
Feat: Avatar.Image can receive any props to the image

### PR DESCRIPTION
**Pull request by kopax**
### Motivation

I wanted to pass `defaultSource` and `resizeMode` but I was not able to.

This will fix the issue

### CODE REVIEW

**Comment by kamui545** [comment]
Glad I saw this, I was about to send a PR.
Same use case, I want to set `defaultSource`

File: src/components/Avatar/AvatarImage.tsx [hunk 1]
```diff
@@ -17,6 +17,10 @@ type Props = {
    * Image to display for the `Avatar`.
    */
   source: ImageSourcePropType;
+  /**
+   * Pass any props to the `Image`
+   */
+  imageProps?: object;
```
Comment (andreialecu) [inline src/components/Avatar/AvatarImage.tsx:23]: Not great to have `object` here. should probably be the `ImageProps` type from react native.

Comment (kopax) [reply src/components/Avatar/AvatarImage.tsx:23]: I am not using typescript but based on : https://github.com/wcandillon/react-native-expo-image-cache/issues/40#issuecomment-444098997 Does it means, replacing object with:

```js
import { ImageProps } from 'react-native';
```
or
```js
import { ImageStyle } from 'react-native';
```

My IDE (intellij) does not allow to import TS source from non typescript project, I just want make sure I am not adding a bug.

Comment (kamui545) [reply src/components/Avatar/AvatarImage.tsx:23]: yup, ImageProps

File: src/components/Avatar/AvatarImage.tsx [hunk 2]
```diff
@@ -17,6 +18,10 @@ type Props = {
    * Image to display for the `Avatar`.
    */
   source: ImageSourcePropType;
+  /**
+   * Pass any image props to the `Image`
+   */
+  imageProps?: ImageProps;
```
Comment (Trancever) [inline src/components/Avatar/AvatarImage.tsx:24]: I would rather see it implemented in the following way:

```
type Props = React.ComponentProps<typeof Image> & {
    // props
}
```

And then we can get `rest` from destructuring and pass it to the `Image` component.

We follow this pattern in other components so would be awesome if we could also use it here.

**Comment by Trancever** [comment]
@kopax Let us know if you have a capacity to work on this or someone else should finish it, thanks!

**Comment by kopax** [comment]
The problem is that I am not using TypeScript myself so even a tiny change is just speculation.

It's better if someone with TS experience takes care of it.

**Comment by Trancever** [comment]
@kamui545 Do you want to fix the remaining issues so we can proceed and merge the PR?

**Comment by kamui545** [comment]
Sure, however I am confused about one of your comment regarding the implementation.

> I would rather see it implemented in the following way:
> 
> ```
> type Props = React.ComponentProps<typeof Image> & {
>     // props
> }
> ```
> 
> And then we can get `rest` props from destructuring and pass it to the `Image` component.
> 
> We follow this pattern in other components so would be awesome if we could also use it here.

`rest` props are already destructured and used on the `View` component that is wrapping the `Image` component.

Are you suggesting we should use `rest` on the `Image` component instead? However, that would not be backward compatible and a breaking change?

**Comment by Trancever** [comment]
@kamui545 Why do you think `rest` is already used? I am looking at the diff and I can't really see it :/

**Comment by kamui545** [comment]
> @kamui545 Why do you think `rest` is already used? I am looking at the diff and I can't really see it :/

PR is too old, please check master: https://github.com/callstack/react-native-paper/blob/53c8e7a255ede4b15f23e4953518525697dd8dea/src/components/Avatar/AvatarImage.tsx#L58

**Comment by kopax** [comment]
If this is solved it can be safely closed. I am still on react-native-paper 3.10.1

**Comment by kamui545** [comment]
> If this is solved it can be safely closed. I am still on react-native-paper 3.10.1

It is not... `rest` is applied to the container, not the image

**Comment by Trancever** [comment]
@kamui545 Please create another PR that is up to date with master.

**Comment by kamui545** [comment]
> @kamui545 Please create another PR that is up to date with master.

Done https://github.com/callstack/react-native-paper/pull/2168

**Pull request closed without merging**